### ISSUE TITLE
Python hosts not being properly picked up

### ISSUE THREAD

**Issue by MatthewMDavis** [neovim/neovim#4535]
- Neovim version: 0.1.2
- Operating system: OSX 10.11.3
- Terminal emulator:  iTerm2 

### Actual behaviour
As described in Issue #3577, I am getting the error message as follows:

`E117: Unknown function: provider#python#Call`

Issue #3577 is concerned solely with a better error message, e.g. 'The neovim package was not found for your python interpreter'. But that's not my problem: my problem is that there unquestionably _is_ a neovim package for the pythons on my sytem, which isn't being found.

I used to think this was being caused by pyenv usage as discussed in Issue #4101, but since I haven't been switching python versions much recently, I completely removed pyenv, and just put python and python3 from Homebrew. Then I did 

```
sudo pip install neovim
sudo pip3 install neovim
```
And added the following to init.vim:
```
let g:python2_host_prog = '/usr/local/bin/python'
let g:python3_host_prog = '/usr/local/bin/python3'
```

If I use `:python "print 'hello'"`, I continue to get the error messages. Same with `:python3 "print('hello')"`.

Here's the output of `:!pip2 show neovim`:

```
Metadata-Version: 2.0
Name: neovim
Version: 0.1.7
Summary: Python client to neovim
Home-page: http://github.com/neovim/python-client
Author: Thiago de Arruda
Author-email: tpadilha84@gmail.com
Installer: pip
License: Apache
Location: /usr/local/lib/python2.7/site-packages
Requires: trollius, msgpack-python, greenlet
Classifiers:
```
The output of `:!pip3 show neovim` parallels the above exactly, so I won't waste space on that. Clearly, there is a pip for each version, and pip knows about the neovim package. So I'm baffled.

**Comment by bfredl** [comment]
requires master neovim.

ping @justinmk would it be feasible to tag 0.1.3 soon?

**Comment by justinmk** [comment]
@bfredl Yes, I'll do it tonight. Sorry for the delay.

**Comment by ninrod** [comment]
Sorry, @bfredl  I did not understand. Do you mean that this is this fixed on master? I'm also having this issue and it's preventing me from using sirver's UltiSnips plugin.

**Comment by bfredl** [comment]
And you do have python-client 0.1.7 I presume?

**Comment by ninrod** [comment]
nvim installed with `brew install neovim`

```txt
~ # nvim --version
NVIM 0.1.2
Build type: RelWithDebInfo
Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/build/config -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/src -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include/luajit-2.0 -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/deps-build/usr/include -I/usr/local/opt/gettext/include -I/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/build/src/nvim/auto -I/tmp/neovim20160215-54286-1aeldc8/neovim-0.1.2/build/include
Compiled by ninrod@ninbook.local

Optional features included (+) or not (-): +acl   +iconv    +jemalloc
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.1.2/share/nvim"

```

Here's the python client version: 

```txt
~ # pip show neovim
---
Metadata-Version: 2.0
Name: neovim
Version: 0.1.7
Summary: Python client to neovim
Home-page: http://github.com/neovim/python-client
Author: Thiago de Arruda
Author-email: tpadilha84@gmail.com
Installer: pip
License: Apache
Location: /usr/local/lib/python2.7/site-packages
Requires: trollius, msgpack-python, greenlet
Classifiers:
~ #
```

invoking nvim with `nvim -u NONE` and running `:py print("foo")` gives me the following

```txt
Error detected while processing function provider#python#Call:
line   18:
no request handler registered for "python_execute"
Press ENTER or type command to continue
```

**Comment by bfredl** [comment]
it seems like neovim 0.1.3 hasn't propagated to brew yet.

**Comment by ninrod** [comment]
this was on `OSX 10.11.4`, but I'm also getting that on GNU/Linux Archlinux: 

```sh
~ # uname -a
Linux vagrant-arch.vagrantup.com 4.4.3-1-ARCH #1 SMP PREEMPT Fri Feb 26 15:09:29 CET 2016 x86_64 GNU/Linux
```

same version of nvim, same version of the python client, same command to invoke nvim (`nvim -u NONE`) same test, `:py print("foo")`, same error.

**Comment by ninrod** [comment]
Oh, ok. @bfredl. If i compile nvim from master, it will work, right? That's what you mean?

**Comment by bfredl** [comment]
yes, or sync arch linux which has 0.1.3 in community.

**Comment by ninrod** [comment]
Ok. I think this prevents any plugin that uses python from working on nvim right know. That's kind of huge.

**Comment by bfredl** [comment]
It prevents them on a specific combination of older versions of neovim and python-client. Latest released version (0.1.3 + 0.1.7) combination is working fine (at least, no one has reported this problem for these versions).

**Comment by bfredl** [comment]
Looks like brew updated. https://github.com/neovim/homebrew-neovim/blob/master/Formula/neovim.rb#L6. Upgrade your neovim installations and it should be fine (no master versions needed).

**Comment by MatthewMDavis** [comment]
I can definitely say that, while completely unaware of all the above, as
soon as I did a 'brew update' and went to 0.1.13, everything worked just
fine. No new configuration required.

Thanks.

**Comment by killtheliterate** [comment]
I resolved this by reinstalling the pip package with `pip install --user neovim`. The extant package must've been installed with `sudo` or something.

**Comment by kaukas** [comment]
0.1.3 + 0.1.7 and still having python3 issues.

This one is Airline specific:
```
Error detected while processing function <SNR>28_on_window_changed[5]..airline#update_statusline[21]..<SNR>29_invoke_funcrefs[2]..airline#util#exec_funcrefs[2]..airline#extensions#
default#apply[5]..<SNR>74_build_sections[5]..<SNR>74_add_section[9]..<SNR>74_get_section:
line    7:
E121: Undefined variable: g:airline_section_a
...
```

After running `UpdateRemotePlugins`:
```
function remote#host#UpdateRemotePlugins[6]..<SNR>26_RegistrationCommands[13]..remote#host#Require[13]..provider#pythonx#Require, line 15
Vim(if):Channel was closed by the client
function remote#host#UpdateRemotePlugins[6]..<SNR>26_RegistrationCommands[13]..remote#host#Require[13]..provider#pythonx#Require, line 22
Failed to load python3 host. You can try to see what happened by starting Neovim with the environment variable $NVIM_PYTHON_LOG_FILE set to a file and opening the generated log file. Also, the host stderr will be available in Neovim log, so it may contain useful information. See also ~/.nvimlog.
remote/host: generated the manifest file in "/Users/user/.vim/.init.vim-rplugin~"
```

`.nvimlog` only says:
```
2015/11/10 05:25:53 [error @ call_set_error:743] 9662 - msgpack-rpc: Channel was closed by the client
```

Can I help debug this issue? NVim is much less usable right now.

**Comment by quozd** [comment]
@kaukas try to reinstall python client. It fixed all my issues after update.

**Comment by kaukas** [comment]
@quozd, I did, no difference. NVim installed via Brew.

```bash
$ nvim --version
NVIM 0.1.3

$ pip2 show neovim
---
Name: neovim
Version: 0.1.7
Location: /Library/Python/2.7/site-packages
Requires: msgpack-python, trollius, greenlet

$ pip3 show neovim
---
Metadata-Version: 2.0
Name: neovim
Version: 0.1.7
Summary: Python client to neovim
Home-page: http://github.com/neovim/python-client
Author: Thiago de Arruda
Author-email: tpadilha84@gmail.com
Installer: pip
License: Apache
Location: /usr/local/lib/python3.5/site-packages
Requires: msgpack-python, greenlet
Classifiers:
```

**Comment by ninrod** [comment]
I can confirm that the issue was resolved upgrading nvim to version 0.1.3.

**Comment by kaukas** [comment]
`UpdateRemotePlugins` fails every time. But the Airline issue I narrowed down to a file that has only this content (*ruby-file.rb*):
```
# vi:filetype=ruby
```

Does it work for others?

**Comment by haifengkao** [comment]
`upgrading nvim to version 0.1.3` is not working.
```
pip uninstall neovim
pip install neovim
```
are not working as well

**Comment by fwalch** [comment]
@haifengkao Try the `:CheckHealth` command. If your problem persists, I think you should open a new issue.. this one is pretty old and seems to have been resolved.

**Comment by haifengkao** [comment]
@fwalch Thanks. I don't have `:CheckHealth`. It seems that the neovim on the homebrew is too old. After I install 0.15, the problem is gone

**Comment by rohanpai** [comment]
Hi I'm also having this problem.  I added `let g:python2_host_prog = '/usr/bin/python'` to my nvimrc

**Comment by david-polak** [comment]
If anyone stumbles upon this. The solution from @rohanpai works with minor alteration `let g:python_host_prog = '/usr/bin/python'`
